pkg/systemd/overview-cards/insights.jsx

```diff
@@ -0,0 +1,172 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+
+import cockpit from "cockpit";
+import * as service from "service.js";
+
+import "./insights.scss";
+
+const _ = cockpit.gettext;
+
+const InsightsIcon = ({ critical, important, moderate, low }) => {
+    const vars = {
+        "--critical": critical,
+        "--important": important,
+        "--moderate": moderate,
+        "--low": low
+    };
+    return (
+        <span className="insights-icon" style={vars}>
+            <span className="insights-icon-critical" />
+            <span className="insights-icon-important" />
+            <span className="insights-icon-moderate" />
+            <span className="insights-icon-low" />
+        </span>);
+};
+
+export class InsightsStatus extends React.Component {
+    constructor() {
+        super();
+        this.state = { };
+    }
+
+    componentDidMount() {
+        this.insights_client_timer = service.proxy("insights-client.timer");
+        this.insights_client_timer.addEventListener("changed", () => this.setState({}));
+
+        const watch = (name, state, parser) => {
+            return cockpit.file(name, { syntax: { parse: parser } }).watch((data, tag, error) => {
+                if (error)
+                    console.warn("Parse error", name, error.toString());
+                this.setState({ [state]: data });
+            });
+        };
+
+        this.id_watch = watch("/var/lib/insights/host-details.json", "id",
+                              data => JSON.parse(data).results[0].id);
+
+        this.hits_watch = watch("/var/lib/insights/insights-details.json", "hits",
+                                data => {
+                                    const json = JSON.parse(data);
```

Comment:
So any exception thrown in these parsers will not crash the page, but just propagate to the `error` watch return argument, right?

Reply:
Yes.